src/file_utils.R

```diff
@@ -182,14 +240,96 @@ filter_csv_obs <- function(outfile, obs_csv, site_ids, obs_start, obs_stop){
+  model_info <- yaml::yaml.load_file(model_out_fl)
+
+  tibble(file = names(model_info), hash = unlist(model_info)) %>% 
+    split_pb_filenames() %>% 
+    mutate(source_filepath = file.path(source_dir, file), out_file = sprintf('%s_%s_%s.csv', exp_prefix, site_id, exp_suffix)) %>% 
+    select(site_id, source_filepath, out_file, hash)
+}
+
+export_pbmtl_df <- function(model_out_fl, exp_prefix, exp_suffix){
+  source_dir <- paste0('../', str_split(dirname(model_out_fl), '/')[[1]][2])
+  stopifnot(dir.exists(source_dir)) # safety check since this is a hacky way to get the dir
+  model_info <- yaml::yaml.load_file(model_out_fl)
+  
+  # need to extract "nhdhr_143418975" from "4_transfer/out/nhdhr_143418975_t|s_nhdhr_143418891_temperatures.feather"
+  tibble(source_filename = names(model_info), hash = unlist(model_info)) %>% rowwise() %>% 
+    mutate(site_id = {str_split(basename(source_filename), '_t\\|s_')[[1]][1]}) %>% ungroup() %>% 
+    mutate(source_filepath = file.path(source_dir, source_filename), out_file = sprintf('%s_%s_%s.csv', exp_prefix, site_id, exp_suffix)) %>% 
+    select(site_id, source_filepath, out_file, hash)
+}
+
+export_mtl_df <- function(site_ids, dir_template, file_pattern, exp_prefix, exp_suffix, dummy){
+  
+  tibble(site_id = site_ids) %>% 
+    mutate(source_dir = sprintf(dir_template, site_id)) %>% rowwise() %>% 
+    mutate(source_filepath = file.path(source_dir, {tibble(file = dir(source_dir)) %>% filter(stringr::str_detect(file, file_pattern)) %>% pull(file)})) %>% 
+    ungroup() %>% mutate(hash = tools::md5sum(source_filepath)) %>% 
+    mutate(out_file = sprintf('%s_%s_%s.csv', exp_prefix, site_id, exp_suffix)) %>% 
+    select(site_id, source_filepath, out_file, hash)
+}
+
+zip_mtl_export_groups <- function(outfile, file_info_df, site_groups,
+                                 export = c('pgmtl_predictions','pgmtl9_predictions')){
+  
+  export <- match.arg(export)
+  
+  model_feathers <- inner_join(file_info_df, site_groups, by = 'site_id') %>%
+    select(-site_id)
+  
+  zip_pattern <- paste0('tmp/', export, '_%s.zip')
+  
+  cd <- getwd()
+  on.exit(setwd(cd))
+  
+  groups <- rev(sort(unique(model_feathers$group_id)))
+  data_files <- c()
+  
+  for (group in groups){
+    zipfile <- sprintf(zip_pattern, group)
+    
+    these_files <- model_feathers %>% filter(group_id == !!group)
+    
+    zippath <- file.path(getwd(), zipfile)
+    
+    if (file.exists(zippath)){
+      unlink(zippath) #seems it was adding to the zip as opposed to wiping and starting fresh...
+    }
+    
+    for (i in 1:nrow(these_files)){
+      fileout <- file.path(tempdir(), these_files$out_file[i])
+      
+      feather::read_feather(these_files$source_filepath[i]) %>% rename(depth = index) %>% 
+        pivot_longer(-depth, names_to = 'date', values_to = 'temp') %>% 
+        mutate(date = as.Date(date)) %>% 
+        pivot_wider(names_from = depth, values_from = temp, names_prefix = 'temp_') %>% 
```

Comment:
Here's the `pivot_wider` I was imagining above. So I do think you would have caught it already if any duplicate site-date-new_depth combos were occurring, because write_csv for a tibble with list columns gives an error:
```
> tibble(depth=c(1,2,3,3,3,4,4,5), temp=1:8) %>% pivot_wider(names_from=depth, values_from=temp, values_fn=mean) %>% write_csv('test.csv')
## ^works
> tibble(depth=c(1,2,3,3,3,4,4,5), temp=1:8) %>% pivot_wider(names_from=depth, values_from=temp) %>% write_csv('test.csv')
Error in stream_delim_(df, path, ..., bom = bom, quote_escape = quote_escape) : 
  Don't know how to handle vector of type list.
In addition: Warning message:
Values are not uniquely identified; output will contain list-cols.
* Use `values_fn = list` to suppress this warning.
* Use `values_fn = length` to identify where the duplicates arise
* Use `values_fn = {summary_fun}` to summarise duplicates 
```
But note that even with the error, a file (just the column names) does get written...

Reply:
Ouch! empty file when we have an error? that's not good. 

Yes, I will add this other code. [Your original code](https://github.com/USGS-CIDA/lake-temperature-neural-networks/blob/master/1_format/src/prepare_pgdl_data.R#L179) does this with a summarize, which I realized after the fact...thought it was redundant with what was done above, but clearly it isn't (e.g., the case where 0.4 and 0.6 exist)